C24_WMDE_Desktop_DE_03

Sperling-0 opened the pull request:
- The banners are based on the CTRL banner of desktop-de-02

Ticket: https://phabricator.wikimedia.org/T371476

Sperling-0 committed:
Prepare campaign for C24_WMDE_Desktop_DE_03

- The banners are based on the CTRL banner of desktop-de-02
- Move the progress bar logic to its folder

Ticket: https://phabricator.wikimedia.org/T371476

Abban commented on banners/desktop/C24_WMDE_Desktop_DE_03/styles/styles.scss:
```diff
@@ -0,0 +1,44 @@
+@use 'src/themes/Svingle/DonationForm/SubComponents/SelectCustomAmountRadio';
+@use 'src/themes/Svingle/DonationForm/SubComponents/SmsBox';
+@use 'src/themes/Svingle/DonationForm/Forms/MainDonationForm';
+@use 'src/themes/Treedip/DonationForm/Forms/UpgradeToYearlyButtonForm' with (
```
Oops, wrong theme

Abban commented on banners/thank_you/components/ProgressBar.vue:
```diff
@@ -12,7 +12,11 @@ interface Props {
 	fillPercentage: number;
 }
 
-const props = defineProps<Props>();
```
Looks like this change sneaked in

Sperling-0 committed:
Implement VAR for C24_WMDE_Desktop_DE_03

- VAR has different donation form styles inspired from English desktop banner

Ticket: https://phabricator.wikimedia.org/T371476

Abban commented on banners/desktop/C24_WMDE_Desktop_DE_03/currentCampaignTimePercentage.ts:
```diff
@@ -0,0 +1,12 @@
+import { CampaignParameters } from '@src/domain/CampaignParameters';
+import TimeRange from '@src/utils/TimeRange';
+
+export function currentCampaignTimePercentage( date: Date, campaignParameters: CampaignParameters ): number {
```
This is for the double progress bar and isn't being used in the banners currently. I think it can be moved into `src/components/ProgressBar`

Abban commented on banners/desktop/C24_WMDE_Desktop_DE_03/messages.ts:
```diff
@@ -0,0 +1,35 @@
+import AlreadyDonatedModal from '@src/components/AlreadyDonatedModal/translations/AlreadyDonatedModal.de';
+import FallbackBanner from '@src/components/FallbackBanner/messages/FallbackBanner.de';
+
+const messages: TranslationMessages = {
```
We should remove the messages that aren't being used from here. The extra ones at the bottom could also be moved into the component messages.

Abban commented on banners/desktop/C24_WMDE_Desktop_DE_03/styles/MainBanner.scss:
```diff
@@ -2,6 +2,9 @@
 
 $banner-height: 357px !default;
 $form-width: 300px !default;
+$padding-top: 10px !default;
```
These are banner specific styles so you can remove these 3 variables and inline the values.

Sperling-0 replied:
Are you sure about this? I feel we need variables here.

Abban commented on src/themes/Treedip/DonationForm/BubbleForm/SelectCustomAmountRadioBubble.scss:
```diff
@@ -2,6 +2,9 @@
 @use '../../variables/colors';
 
 $height: 23px !default;
+$border-radius: 8px !default;
```
These could also be removed and the values put inline.

Abban commented on banners/desktop/C24_WMDE_Desktop_DE_03/styles/stylesVar.scss:
```diff
@@ -0,0 +1,62 @@
+@use 'src/themes/Svingle/ProgressBar/DoubleProgressBar';
+@use 'src/themes/Svingle/DonationForm/DonationForm';
+@use 'src/themes/Svingle/DonationForm/MultiStepDonation';
+@use 'src/themes/Treedip/DonationForm/BubbleForm/SelectGroupBubble' with (
```
These imports are pointing to the wrong theme.

Abban reviewed (changes requested):
It's mostly perfect, I spotted some unused things that could be moved or deleted.

You imported the wrong theme files for the bubble forms, you should import the Svingle ones and copy the css changes you made in Treedip to there without using variables. Then revert the Treedip changes.
